src/js/pages/chat.js

```diff
@@ -325,6 +327,77 @@ module.exports = BasePage.extend({
         }
         return false;
     },
+    handleMediaClick: function(e){
+        e.preventDefault();
+        var self = this;
+        console.log(this.model.topResource);
+        var file;
+        if (e.dataTransfer) {
+            file = e.dataTransfer.files[0];
+        } else if (e.target.files) {
+            file = e.target.files[0];
+        } else {
+            return;
+        }
+
+        if (file.type.match('image.*')) {
+            var downloadURL = URL.createObjectURL(file);
+            console.log('Offering:', file.name, file.size, file.type, file.lastModifiedDate,  downloadURL);
+            
+            var fileTracker = new FileReader();
+          
+            fileTracker.onload = function () {
+            console.log('file', file.name, file.size, file.type, file.lastModifiedDate);
+            var data =  this.result;        
+            var b64Data = data.split(',')[1];
+            var id = crypto.createHash('sha1').update(atob(b64Data)).digest('hex');   
+            client.use(shareMedia);
```

Comment:
Move it to app.ts - https://github.com/ForNeVeR/Kaiwa/blob/master/src/js/app.ts#L110